core/storage/btree.rs

```diff
@@ -9715,4 +9831,357 @@ mod tests {
+        for i in 1..=num_records {
+            let exists = run_until_done(|| cursor.exists(&Value::Integer(i)), &pager).unwrap();
+            assert!(!exists, "Record {i} should not exist after clearing");
+        }
+    }
+
+    #[test]
+    pub fn test_clear_btree_reinsertion_works() {
+        let (pager, root_page, _, _) = empty_btree();
+        let num_columns = 5;
+        let mut cursor = BTreeCursor::new_table(None, pager.clone(), root_page, num_columns);
+
+        // Insert records
+        let num_records = 50;
+        for i in 1..=num_records {
+            let regs = &[Register::Value(Value::Integer(i))];
+            let record = ImmutableRecord::from_registers(regs, regs.len());
+            let key = BTreeKey::new_table_rowid(i, Some(&record));
+            run_until_done(|| cursor.insert(&key, false), &pager).unwrap();
+        }
+
+        // Clear the table
+        let cell_count = run_until_done(|| cursor.clear_btree(), &pager).unwrap();
+        // Following SQLite's behavior, we count cells on all leaf pages.
+        // For table B-trees, this means counting table leaf cells.
+        assert_eq!(
+            cell_count, num_records as u64,
+            "Should return number of table leaf cells cleared"
+        );
+
+        // Re-insert records after clearing
+        for i in 1..=num_records {
+            let regs = &[Register::Value(Value::Integer(i * 10))]; // Different values
+            let record = ImmutableRecord::from_registers(regs, regs.len());
+            let key = BTreeKey::new_table_rowid(i, Some(&record));
+            run_until_done(|| cursor.insert(&key, false), &pager).unwrap();
+        }
+
+        // Verify re-inserted records exist and have correct values
+        for i in 1..=num_records {
+            let exists = run_until_done(|| cursor.exists(&Value::Integer(i)), &pager).unwrap();
+            assert!(exists, "Re-inserted record {i} should exist");
+
+            let seek_result = run_until_done(
+                || cursor.seek(SeekKey::TableRowId(i), SeekOp::GE { eq_only: true }),
+                &pager,
+            )
+            .unwrap();
+            assert_eq!(
+                seek_result,
+                SeekResult::Found,
+                "Should find re-inserted record {i}",
+            );
+
+            let record = run_until_done(|| cursor.record(), &pager).unwrap();
+            let value = record.unwrap().get_value(0).unwrap();
+            assert_eq!(
+                value,
+                RefValue::Integer(i * 10),
+                "Re-inserted record {i} should have correct value",
+            );
+        }
+    }
+
+    #[test]
+    pub fn test_clear_btree_multi_page_table() {
+        let (pager, root_page, _, _) = empty_btree();
+        let num_columns = 5;
+        let mut cursor = BTreeCursor::new_table(None, pager.clone(), root_page, num_columns);
+
+        // Insert enough records to create multiple pages (interior + leaf pages)
```

Comment:
This test should definitely be duplicated for index btrees too, since interior index cells have payloads

Reply:
ok.  Same setup, I just call `new_index` instead, right ?